### PR TITLE
Export new employees to NetSuite

### PR DESCRIPTION
**Pull request by jferris**
* Adds an "Export Now" button to fully-connected NetSuite accounts
* Creates NetSuite employees for each new Namely profile
* Stores the NetSuite internalId field to track exported profiles
* Skips already-exported profiles
* Displays results from exported employees

https://trello.com/c/IiL9WOyi

### CODE REVIEW

File: app/models/net_suite/client.rb [hunk 1]
```diff
@@ -1,10 +1,16 @@
 module NetSuite
   class Client
     BASE_URL = "https://api.cloud-elements.com/elements/api-v2"
+    GENDER_MAP = {
+      "Male" => "_male",
```
Comment (base10) [inline app/models/net_suite/client.rb]: I take it the underscore is a NetSuite data requirement?

File: spec/features/user_visits_dashboard_spec.rb [hunk 2]
```diff
@@ -206,7 +207,12 @@
           "dashboards.show.missing_namely_field",
           name: "netsuite_id",
         )
+        expect(page).to have_export_button
       end
     end
   end
+
+  def have_export_button
```
Comment (jferris) [inline spec/features/user_visits_dashboard_spec.rb]: This comment is incorrect and is something we're discussing on Hound.

File: app/models/net_suite/client.rb (hunk 1 above)
Comment (jferris) [inline app/models/net_suite/client.rb]: Yes, these are the enumeration values in the NetSuite API. I don't know why they used underscore prefixes.

File: app/models/net_suite/export.rb [hunk 3]
```diff
@@ -0,0 +1,61 @@
+module NetSuite
+  class Export
+    def initialize(namely:, net_suite:)
+      @namely = namely
```
Comment (base10) [inline app/models/net_suite/export.rb]: Thoughts on just passing profiles here vs. the namely connection object?

File: spec/features/user_exports_to_net_suite_spec.rb [hunk 4]
```diff
@@ -0,0 +1,76 @@
+require "rails_helper"
+
+feature "user exports to net suite" do
+  scenario "successfully" do
+    user = create(:user)
+    stub_request(:get, %r{https://.*.namely.com/api/v1/profiles\?.*}).
+      to_return(status: 200, body: <<-JSON.squish)
```
Comment (base10) [inline spec/features/user_exports_to_net_suite_spec.rb]: Moveable to a fixture file?

File: spec/models/net_suite/client_spec.rb [hunk 5]
```diff
@@ -68,4 +68,50 @@
...
+  describe "#create_employee" do
+    context "on HTTP success" do
+      it "returns successful data" do
+        employee = { internalId: "1949" }
+        stub_request(
+          :post,
+          "https://api.cloud-elements.com/elements/api-v2/hubs/erp/employees"
+        ).
+          with(
+            body: {
```
Comment (base10) [inline spec/models/net_suite/client_spec.rb]: I take it we don't want to use a fixture here to keep what we're testing for front-and-center, yes?

Comment (jferris) [inline spec/models/net_suite/client_spec.rb]: Yes, in this case, I think it's helpful to have the relevant data in the test.

File: app/models/net_suite/export.rb (hunk 3 above)
Comment (jferris) [inline app/models/net_suite/export.rb]: Updated in c42b22b.

File: spec/features/user_exports_to_net_suite_spec.rb (hunk 4 above)
Comment (jferris) [inline spec/features/user_exports_to_net_suite_spec.rb]: Done in dfb19c0.

File: app/models/net_suite/export.rb (hunk 3 above)
Comment (base10) [inline app/models/net_suite/export.rb]: Nice, I like that a lot.

File: spec/features/user_exports_to_net_suite_spec.rb (hunk 4 above)
Comment (base10) [inline spec/features/user_exports_to_net_suite_spec.rb]: :+1:

**Comment by base10** [comment]
LGTM, pending decision on https://github.com/namely/connect/pull/112#discussion_r33050837

**Pull request closed without merging**